lib/src/local_working_copy.rs

```diff
@@ -1909,13 +1914,59 @@ impl TreeState {
                 continue;
             }
 
-            // Create parent directories no matter if after.is_present(). This
-            // ensures that the path never traverses symlinks.
-            let Some(disk_path) = create_parent_dirs(&self.working_copy_path, &path)? else {
-                changed_file_states.push((path, FileState::placeholder()));
-                stats.skipped_files += 1;
-                continue;
+            let adjusted_working_copy_path;
+            let adjusted_diff_file_path = if let Some(ref prev_path) = prev_created_path {
+                // Obtain the common prefix between these paths
+                path_prefix = RepoPathBuf::root();
+                path_prefix.extend(
+                    prev_path
+                        .components()
+                        .zip(path.components())
+                        .take_while(|(prev_comp, this_comp)| prev_comp == this_comp)
+                        .map(|(comp, _)| comp),
+                );
+
+                adjusted_working_copy_path = path_prefix.to_fs_path(self.working_copy_path())?;
+
+                path.strip_prefix(&path_prefix)
+                    .expect("path should start with common prefix")
```

Comment:
Can you extract this to a helper function? I think something like this can be added to `RepoPath` type or `repo_path` module:

```rust
fn split_common_prefix(&'a RepoPath, &RepoPath) -> (&'a RepoPath, &'a RepoPath) {
    let prefix_len = ...;
    let common = RepoPath::new_uncheckdd(path.value[..prefix_len]);
    let remainder = RepoPath::new_uncheckdd(path.value[prefix_len + 1..]); // XXX need bounds checking?
    (common, remainder)
}
```